### PR TITLE
[UTOPIA-767]add disclosure tab

### PR DESCRIPTION
**Pull request by binggaofw**
# Description

This PR includes the following proposed change(s):

- add collection, use and disclosure tab

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/frontend/src/routes/router.tsx [hunk 1]
```diff
@@ -44,6 +45,7 @@ const Router = () => {
         <Route path="/pia/intake/:id/:title" element={<PIADetailPage />} />
       </Route>
       <Route path="/not-authorized" element={<Unauthorized />} />
+      <Route path="/tab1" element={<PIACollectionUseAndDisclosure />} />
```
Comment (binggaofw) [inline src/frontend/src/routes/router.tsx:48]: this is just for test purposes, I will delete after the pr got approve

File: src/frontend/src/components/public/PIACollectionUseAndDisclosure/index.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  label: string;
+  filed: string;
+  value: string;
+  id: string;
+}
+
+const PIACollectionUseAndDisclosure = () => {
+  const navigate = useNavigate();
+  const [disclosure, setDisclosure] = useState('');
+  const [MPOCommentsDisclosure, setMPOCommentsDisclosure] = useState('');
```
[the remark this answers is not in the document]
Comment (binggaofw) [reply src/frontend/src/components/public/PIACollectionUseAndDisclosure/index.tsx:20]: yes, i will change it.

File: src/frontend/src/components/public/PIACollectionUseAndDisclosure/index.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+  return (
+    <div className="bcgovPageContainer wrapper ">
+      <h1 className="results-header">{Messages.Headings.Title.en}</h1>
+      <span>{Messages.Headings.Subtitle.en}</span>
+      <h2 className="pt-3 pb-3">{Messages.WorkThroughDetails.Title.en}</h2>
+      <section className="card">
+        <div className="container">
+          <div className="row">
+            <div>
+              <table className="data-table__container" id="tab_logic">
```
Comment (binggaofw) [inline src/frontend/src/components/public/PIACollectionUseAndDisclosure/index.tsx:96]: This list will be replace by my other pr( additional risk tab), I create a list component

File: src/frontend/src/components/public/PIASideNav/index.tsx [hunk 4]
```diff
@@ -34,7 +35,12 @@ const PiaSideNav = (personal_information: PI) => {
   const sideElement = enableSideNavList(
     personal_information.personal_information,
   );
-  return <NavBar pages={sideElement} CSSclass="sidenav" />;
+  return (
+    <>
+      <NavBar pages={sideElement} CSSclass="sidenav" />
+      <Outlet />
```
Comment (Bottle7) [inline src/frontend/src/components/public/PIASideNav/index.tsx:41]: What is Outlet doing here?

**Comment by binggaofw** [comment]
right now if you want to test it, you only can change the URL in the browser, Like this, http://localhost:8080/pia/**:id**/disclosure/edit to any existing pia intake form id in your local database. You can edit and auto-save the data
http://localhost:8080/pia/19/disclosure/edit

File: src/frontend/src/routes/router.tsx [hunk 5]
```diff
@@ -115,6 +115,10 @@ const Router = () => {
             </Route>
           </Route>
         </Route>
+        <Route path="/pia-result" element={<PIAIntakeResultsPage />} />
+        <Route path="/pia-list" element={<PIAList />} />
+        <Route path="/pia-list/*" element={<PIAList />} />
+        <Route path="/pia/intake/:id/:title" element={<PIADetailPage />} />
```
Comment (kushal-arora-fw) [inline src/frontend/src/routes/router.tsx:121]: This is no longer needed

File: src/frontend/src/components/common/Navbar/index.tsx [hunk 6]
```diff
@@ -1,12 +1,13 @@
...
-import { NavLink } from 'react-router-dom';
+import { NavLink, useParams } from 'react-router-dom';
 import { INavbarPages } from './interfaces';
 
 function NavBar({
   pages,
   CSSclass = 'navbar-container wrapper',
 }: INavbarPages): ReactElement {
   const currentPath = window.location.pathname;
+  const { id } = useParams();
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/common/Navbar/index.tsx:10]: unrequired change

Comment (binggaofw) [reply src/frontend/src/components/common/Navbar/index.tsx:10]: remove them

File: src/frontend/src/routes/router.tsx (hunk 5 above)
Comment (binggaofw) [reply src/frontend/src/routes/router.tsx:121]: removed too

File: src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/index.tsx [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+  const [rows, setRows] = useState<Array<InputTextProps[]>>(
+    steps.map((step, i) => [
+      { label: `Step ${i}`, value: step.drafterInput, id: 'one' },
+      { value: step.mpoInput, id: 'two' },
+      { value: step.foippaInput, id: 'three' },
+      { value: step.OtherInput, id: 'four' },
+    ]),
+  );
+
+  const addRow = () => {
```
Comment (Bottle7) [inline src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/index.tsx:90]: I feel like this method, along with the 'removeRow' method, should be encapsulated within the List component itself. Please correct me if I am wrong, but this way of doing things will require us to redefine 'addRow' and 'removeRow' for every tab that uses the List.

Comment (binggaofw) [reply src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/index.tsx:90]: Yes, you are right, already create a (sub-task)[https://apps.itsm.gov.bc.ca/jira/browse/UTOPIA-854]

File: src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/messages.ts [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    Subtitle: {
+      en: `This section will help you to identify the legal authority for collecting, using, and disclosing personal information and to confirm that all personal information elements are necessary for the purpose of the initiative.`,
+    },
+  },
+
+  WorkThroughDetails: {
+    Title: {
+      en: `Step-by-step walkthrough`,
+    },
+    DescriptionColumnOne: {
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/messages.ts:16]: change naming convention from columnOne, etc to more meaningful name

File: src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/messages.ts [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+      en: `Describe the way personal information moves through your initiative step by step as if you were explaining it to someone who does not know about your initiative.`,
+    },
+    DescriptionColumnTwo: {
+      en: `Collection, use, disclosure  (for MPO use only)`,
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/messages.ts:20]: (for MPO use only) -- is not bold in designs and should be treated differently

File: src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/index.tsx [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+    Messages.WorkThroughDetails.DescriptionColumnOne.en,
+    Messages.WorkThroughDetails.DescriptionColumnTwo.en,
+    Messages.WorkThroughDetails.DescriptionColumnThree.en,
+    Messages.WorkThroughDetails.DescriptionColumnFour.en,
+  ];
+  return (
+    <>
+      <h1 className="results-header">{Messages.Headings.Title.en}</h1>
+      <span>{Messages.Headings.Subtitle.en}</span>
+      <h2 className="pt-3 pb-3">{Messages.WorkThroughDetails.Title.en}</h2>
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/index.tsx:152]: refer as we did in the other tab?

File: src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/index.tsx [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+          columnsName={columnsName}
+          handleOnChange={handleOnChange}
+          addRow={addRow}
+          removeRow={removeRow}
+          enableRemove={true}
+          sourceTab="disclosure"
+        />
+      </section>
+
+      <h2 className="pt-4 pb-4">{Messages.CollectionNotice.Title.en}</h2>
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/index.tsx:165]: add appropriate spacing as the Section above

File: src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/index.tsx [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+
+      <section className="card pt-5 px-5">
+        <h2>{Messages.CollectionNotice.PartOne.Title.en}</h2>
+        <p>{Messages.CollectionNotice.PartOne.Description.en}</p>
+        <div className="richText" id="drafterDisclosure">
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/index.tsx:170]: update to label and sub implementation as discussed

File: src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/index.tsx [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+              setCollectionNotice({
+                mpoInput: MPOCommentsDisclosure,
+                drafterInput: disclosure,
+              });
+              stateChangeHandler(collectionNotice, 'collectionNotice');
+            }}
+          />
+        </div>
+        <h2 className="pt-5">{Messages.CollectionNotice.PartTwo.Title.en}</h2>
+        <div className="richText pb-4" id="MPOCommentsDisclosure">
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/index.tsx:186]: same as above - label impl and appropriate spacing

File: src/frontend/src/components/common/List/index.tsx [hunk 14]
```diff
@@ -36,7 +39,13 @@ const List = ({
           <thead>
             <tr>
               {columnsName.map((column, index) => (
-                <th key={index}>{column}</th>
+                <th key={index}>
+                  {column.split('"').map((name, idx) => (
+                    <div className={idx === 0 ? '' : 'no-bold'} key={name}>
+                      {name}
+                    </div>
+                  ))}
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/common/List/index.tsx:47]: I don't think splitting with string `"` is the right way to add anchor tags

Comment (kushal-arora-fw) [reply src/frontend/src/components/common/List/index.tsx:47]: Refer Implementation here
src/frontend/src/components/public/PIAFormTabs/intake/helper/messages.ts
src/frontend/src/components/public/PIAFormTabs/intake/index.tsx

Comment (kushal-arora-fw) [reply src/frontend/src/components/common/List/index.tsx:47]: image

File: src/frontend/src/sass/_font.scss [hunk 15]
```diff
@@ -13,3 +13,8 @@
 .text__line-height--2 {
   line-height: 2;
 }
+
+.section__question-hint {
+  color: $text-gray;
+  ;
+}
```
Comment (kushal-arora-fw) [inline src/frontend/src/sass/_font.scss:20]: missing End of Line

File: src/frontend/src/sass/_font.scss [hunk 16]
```diff
@@ -13,3 +13,8 @@
 .text__line-height--2 {
   line-height: 2;
 }
+
+.section__question-hint {
+  color: $text-gray;
+  ;
```
Comment (kushal-arora-fw) [inline src/frontend/src/sass/_font.scss:19]: unnecessary semi colon!

File: src/frontend/src/sass/_table.scss [hunk 17]
```diff
@@ -39,8 +47,8 @@
           width: 50ch;
         }
 
-        & > td:nth-child(5) {
-          padding: 0;
+        & > td.no-padding {
+           padding: 0;
```
Comment (kushal-arora-fw) [inline src/frontend/src/sass/_table.scss:51]: unnecessary extra spacing

File: src/frontend/src/sass/_font.scss (hunk 16 above)
Comment (binggaofw) [reply src/frontend/src/sass/_font.scss:19]: remove it

**Review by kushal-arora-fw — APPROVED** [review]

**Pull request merged**